nengo/tests/test_pytest.py

```diff
@@ -10,31 +12,91 @@ def test_seed_fixture(seed):
     assert i == 1832276344
 
 
-def test_unsupported(testdir):
+@pytest.mark.parametrize("xfail", (True, False))
+def test_unsupported(xfail, testdir):
+    testdir.makefile(".py", test_file="""
+        import pytest
+
+        @pytest.mark.parametrize("param", (True, False))
+        def test_unsupported(param):
+            assert param
+
+        @pytest.mark.parametrize("param", (True, False))
+        def test_unsupported_all(param):
+            assert False
+
+        def test_supported():
+            assert True
+        """)
+
+    testdir.makefile(".ini", pytest="""
+        [pytest]
+        nengo_test_unsupported =
+            test_file.py:test_unsupported[False]
+                "One unsupported param
+                with multiline comment"
+            test_file.py:test_unsupported_all*
+                "Two unsupported params
+                with multiline comment"
+        """)
+
+    testdir.makefile(".py", conftest="""
+        from nengo.conftest import pytest_runtest_setup, pytest_configure
+        """)
+
+    args = "-p nengo.tests.options -rsx".split()
+    if xfail:
+        args.append("--unsupported")
+    output = testdir.runpytest(*args)
+
+    output.stdout.fnmatch_lines_random([
+        "*One unsupported param with multiline comment",
+        "*Two unsupported params with multiline comment",
+    ])
+
+    outcomes = output.parseoutcomes()
+    if xfail:
+        assert outcomes["xfailed"] == 3
+        assert "skipped" not in outcomes
+    else:
+        assert outcomes["skipped"] == 3
+        assert "xfailed" not in outcomes
+    assert "failed" not in outcomes
+    assert outcomes["passed"] == 2
+
+
+def test_pyargs(testdir):
+    # create a simulator that immediately fails (we don't actually want
+    # to run all the tests)
     testdir.makeconftest(
         """
         import nengo.conftest
 
         class MockSimulator(object):
-            unsupported = [('*', 'mock simulator')]
+            pass
 
         nengo.conftest.TestConfig.Simulator = MockSimulator
         """)
-    outcomes = testdir.runpytest(
-        "-p", "nengo.tests.options", "--pyargs", "nengo",
-    ).parseoutcomes()
-    assert outcomes["skipped"] > 350
-    assert outcomes["deselected"] > 700
-    assert "passed" not in outcomes
-    assert "failed" not in outcomes
 
-    outcomes = testdir.runpytest(
-        "-p", "nengo.tests.options", "--pyargs", "nengo", "--unsupported",
-    ).parseoutcomes()
+    # mark all the tests as unsupported
+    testdir.makefile(".ini", pytest="""
+        [pytest]
+        nengo_test_unsupported =
+            *
+                "Using mock simulator"
+        """)
+
+    try:
+        outcomes = testdir.runpytest(
+            "-p", "nengo.tests.options",
+            "--pyargs", "nengo",
+            "--unsupported",
```

Comment:
This is why we have the dummy Simulator from above, so that we're not actually simulating anything we just fail as soon as we get to the Simulator part of the test.  I was thinking that we'd want the tests to actually run, but I'm also fine with just skipping them.  In that case we should be able to remove the Simulator from above (since we won't be running the test, we don't need it to immediately fail, we can just use the regular Simulator).

Reply:
It takes about 15 seconds with `--unsupported` and around 2.5 without on my machine. Some tests have a bit of stuff before `Simulator` is actually instantiated, for example if we're creating reference data to compare to.

Reply:
Yeah that's why I was thinking it'd be better to xfail, so that we're actually running part of the tests at least.  But I don't have anything in particular in mind that I was thinking that'd catch, it was just a general precaution.  So I'm fine skipping them.